### PR TITLE
fix(ipc): scope active-user IPC routing to root CLI main requests

### PR DESCRIPTION
**Pull request by fufesou**
## Summary

Fix the command line when running with `sudo`. The regression is introduced in https://github.com/rustdesk/rustdesk/pull/14671 

```bash
rustdesk --get-id
sudo rustdesk --get-id

sudo rustdesk --set-id a1234567
rustdesk --get-id

sudo rustdesk --password <pass1>
sudo rustdesk --password <pass2>

sudo rustdesk --set-unlock-pin 1234

sudo rustdesk --option custom-rendezvous-server
sudo rustdesk --option custom-rendezvous-server 192.168.1.14
sudo rustdesk --option custom-rendezvous-server

sudo rustdesk --option api-server
sudo rustdesk --option api-server 192.168.1.13
sudo rustdesk --option api-server

sudo rustdesk --config <config1>
sudo rustdesk --option custom-rendezvous-server

sudo rustdesk --config <config2>
sudo rustdesk --option custom-rendezvous-server

sudo rustdesk --assign --token <token> --device_name 'dev name1'
```

## Desc

On Linux/macOS, when an installed RustDesk CLI is run as root, only normal main IPC requests (`postfix == ""`) from management commands are routed to the matching `--server` owner:

- one matching `--server`: use that uid
- multiple `--server` processes: use the active user's server
- Linux Wayland login screen: prefer the root `--server`
- still ambiguous: return an error instead of guessing

On Windows, main IPC now accepts elevated administrator peers, which keeps commands such as `--password` working on the login screen. Rejected main IPC logs also include the peer elevation state.

Other IPC paths keep their existing behavior, including `_service`, `_cm`, `_url`, `_uinput_*`, and portable-service IPC.

## Testing

- [x] macOS, Linux, one user logged in
- [x] macOS, Linux, login screen, no users logged in
- [x] macOS, Linux, login screen, one user logged in
- [x] macOS, Linux, directly run, no `--server`
- [x] Linux, two users logged in
- [x] Linux, login screen, two users logged in
- [x] Windows on user logged in
- [x] Windows, A (not admin) logged in, run as the Administrator
- [x] Windows, login screen, no users logged in
- [x] Windows, two users logged in
- [x] Windows, login screen, two users logged in

### History issue

- macOS, two users logged in. There are two `--server` processes. The command line only changes the configs of the active user.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved IPC routing so privileged management CLI commands running as root use the correct per-user service when installed and applicable.
  * Windows IPC authorization now recognizes elevated peers, accepts them in more scenarios, and includes elevation info in rejected-connection logs.
  * More robust multi-user IPC server selection on Linux/macOS to handle ambiguous or missing service instances.

* **Tests**
  * Added tests for CLI command classification and UID/server selection edge cases.

* **Chores**
  * Updated common library pointer.

[Review Change Stack](https://app.coderabbit.ai/change-stack/rustdesk/rustdesk/pull/15058?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/ipc.rs [hunk 1]
```diff
@@ -1204,6 +1240,93 @@ async fn connect_with_path(ms_timeout: u64, path: &str) -> ResultType<Connection
...
+        }
+        let Ok(process_path) = std::fs::canonicalize(process.exe()) else {
+            continue;
+        };
+        if process_path != current_exe_path {
+            continue;
+        }
+        let Some(uid) = process.user_id().map(|uid| **uid as u32) else {
+            bail!("Failed to read --server process uid");
+        };
```
[the remark this answers is not in the document]
Comment (fufesou) [reply src/ipc.rs:1303]: I’d prefer to keep this as a hard error.

This code is only used for CLI management command routing. At this point, we have already matched a RustDesk `--server` process from the same executable, and `--server` is the key target process for the command. It is not expected to exit frequently. If it does race during this short enumeration window and we cannot read its uid, failing the command is clearer than silently skipping it and possibly selecting another process.

The user can retry the CLI command in that case. I added a short comment to make this intentional.

**Comment by fufesou** [comment]
@coderabbitai review

**Comment by fufesou** [comment]
@coderabbitai review

**Comment by fufesou** [comment]
@coderabbitai review
